lib/db.js

```diff
@@ -349,46 +380,46 @@ module.exports = (
 
   DB.prototype.devices = function (uid) {
     log.trace({ op: 'DB.devices', uid: uid })
+    const promises = [
+      this.pool.get('/account/' + uid + '/devices')
+    ]
 
-    return this.pool.get('/account/' + uid + '/devices')
-      .then(
-        function (body) {
-          return body.map(function (item) {
-            return {
-              id: item.id,
-              sessionToken: item.sessionTokenId,
-              lastAccessTime: marshallLastAccessTime(item.lastAccessTime, uid, item.email),
-              name: item.name,
-              type: item.type,
-              pushCallback: item.callbackURL,
-              pushPublicKey: item.callbackPublicKey,
-              pushAuthKey: item.callbackAuthKey,
-              uaBrowser: item.uaBrowser,
-              uaBrowserVersion: item.uaBrowserVersion,
-              uaOS: item.uaOS,
-              uaOSVersion: item.uaOSVersion,
-              uaDeviceType: item.uaDeviceType
-            }
-          })
-        },
-        function (err) {
-          if (isNotFoundError(err)) {
-            throw error.unknownAccount()
+    if (this.redis) {
+      promises.push(this.redis.getAsync(uid))
+    }
+    return P.all(promises)
+      .spread((devices, redisTokens) => {
+        const redisSessionTokens = redisTokens ? JSON.parse(redisTokens) : []
```

Comment:
Could you please add a short doc string that explains what the 5 lines below are doing. This is for future readers of this code